Sources/XMTPiOS/Client.swift

```diff
@@ -172,13 +233,36 @@ public final class Client: Sendable {
 			rustClient: client
 		)
 
-		return try Client(address: address, privateKeyBundleV1: v1Bundle, apiClient: apiClient)
+		let v3Client = try await initV3Client(
+			address: address,
+			options: options,
+			source: .static,
+			privateKeyBundleV1: v1Bundle
+		)
+
+		if let v3Client, let textToSign = v3Client.textToSign() {
+			guard let signingKey else {
+				throw ClientError.creationError("No v3 keys found, you must pass a SigningKey in order to enable alpha MLS features")
+			}
+
+			let signature = try await signingKey.sign(message: textToSign)
+			try await v3Client.registerIdentity(recoverableWalletSignature: signature.rawData)
+		}
```

Comment:
Since this check is required for V3 client creation I think it should move into the above method. And the above method should take an optional signer.

Reply:
I also think the logic here is incorrect if there is no text to sign that just means a V3 identity already exists but you still need to call registerIdentity on it. If there is text to sign you sign it and call register identity. Otherwise then error.

```
        if (v3Client.textToSign() == null) {
            v3Client.registerIdentity(null)
        } else if (account != null) {
            v3Client.textToSign()?.let {
                v3Client.registerIdentity(account.sign(it))
            }
        } else {
            Log.i(TAG, "No signer passed but signer was required.")
        }
```